x/btcstaking/keeper/grpc_query.go

```diff
@@ -67,12 +67,27 @@ func (k Keeper) FinalityProvider(c context.Context, req *types.QueryFinalityProv
 	}
 
 	ctx := sdk.UnwrapSDKContext(c)
+	currBlockHeight := uint64(ctx.BlockHeight())
+
 	fp, err := k.GetFinalityProvider(ctx, key)
 	if err != nil {
+		// Try in the btcstkconsumer module
+		if k.BscKeeper.HasConsumerFinalityProvider(ctx, fpPK) {
```

Comment:
if we are going to have 2 different stores with FPs we should return the same structure/interface and have a function to query a FP by their fpPK to avoid handling in the error case

Reply:
Suggest we iterate over this part based on the feedbacks from web team later. It's not clear to me whether this interface should dedicate to Babylon FP or be a unified one for both Babylon/consumer FP. Wdyt?